### DISCUSSION TITLE
Rename or namespace the `Teams` module

### DISCUSSION THREAD

**Discussion by robjacoby** [rubyatscale/code_teams#4] [Ideas]
`Teams` is quite a generic term, and unfortunately it already exists as a domain/package in our monolith as a top-level namespace. This is preventing us from using both this gem, as well as https://github.com/rubyatscale/code_ownership which has a dependency on this gem.

I would like to propose that either the name of the top-level constant `Teams` is changed to something less generic, or ideally, it is just namespaced with something less generic, making it more specific to the problem this gem is solving.

Keen to hear your thoughts!

**Comment by alexevanczuk** [comment]
@robjacoby Thanks for starting this discussion. I chatted with my team and we agree with you – `Teams` is way too generic.

Here are some ideas for better names:
- `CodeTeams`
- `TeamOrganizations`
- `DeveloperTeams`
- `EngineeringTeams`
- `TeamRepository`

I'll probably go with `CodeTeams` but open to other suggestions.

**Reply by robjacoby** [reply]
`CodeTeams` works for my use case, although I'm wondering whether you have plans to utilize the `RubyAtScale` namespace or similar? Perhaps grouping the gems within a single top-level namespace would give you the freedom to use the more generic terms (like `Teams`) within them.

Just a thought 😄 

Thanks for entertaining the idea @AlexEvanczuk!

**Reply by alexevanczuk** [reply]
In progress PR here: https://github.com/rubyatscale/bigrails-teams/pull/5
I will follow up once it lands.

Yeah I've been thinking about that top-level namespace. It's a bit off a toss-up for me, because the client of the gem would still need to write `RubyAtScale::Teams` which to me is a bit more unwieldy than `CodeTeams`, and has portions not related to the specific domain (i.e. `RubyAtScale`). I also like the mindset that not having a global namespace sets, which is that while the gems in this org ideally should work nicely together (i.e. they should all have well-modularized boundaries), they can also be used independently or in conjunction with other/different tools. Definitely possible we will change course on this at some point though, cc @ngan 

Wanted to thank you again for bringing this up and being an early adopter and providing that very critical early feedback, it helps me/Gusto be better at open source :)

**Reply by alexevanczuk** [reply]
@robjacoby This transition is done! You should now be able to use the `code_teams` (namespaces under `CodeTeams`) gem. The other gems in the toolchain (i.e. within this org) should also have a newly published version using `code_teams`. Please let me know of any other issues you have! Thanks again for opening up this issue.

**Reply by robjacoby** [reply]
Thanks @AlexEvanczuk! I really appreciate you taking the time to consider this and make the updates so quickly!
